### PR TITLE
Remove `RequestParts::take_extensions`

### PR DESCRIPTION
**Pull request by davidpdrsn**
Same story as https://github.com/tokio-rs/axum/pull/698 but this time for extensions.

Removes `RequestParts::take_extensions` and `impl FromRequest for Extensions` meaning `Extensions` is no longer an extractor. Been thinking before that it probably didn't have much utility in the first place since all you can do is [`get`](https://docs.rs/http/latest/http/struct.Extensions.html#method.get) types, which `extract::Extension` can also do. Users can still use `RequestParts::extensions_mut` however to `std::mem::take` them if they really need to.

## TODO

- [x] Changelog

### CODE REVIEW

File: axum-core/src/extract/mod.rs [hunk 1]
```diff
@@ -108,52 +108,38 @@ impl<B> RequestParts<B> {
...
-            extensions: Some(extensions),
+            extensions,
             body: Some(body),
         }
     }
 
     /// Convert this `RequestParts` back into a [`Request`].
     ///
-    /// Fails if
+    /// Fails if The request body has been extracted, that is [`take_body`] have
```
Comment (jplatte) [inline axum-core/src/extract/mod.rs:118]:
```suggestion
    /// Fails if The request body has been extracted, that is [`take_body`] has
```

File: axum-core/src/macros.rs [hunk 2]
```diff
@@ -26,12 +26,6 @@ macro_rules! define_rejection {
         }
 
         impl std::error::Error for $name {}
-
-        impl Default for $name {
-            fn default() -> Self {
-                Self
-            }
-        }
```
Comment (jplatte) [inline axum-core/src/macros.rs:34]: This change seems unrelated.

Comment (davidpdrsn) [reply axum-core/src/macros.rs:34]: Since `BodyAlreadyExtracted` is the only rejection left in core that didn't wrap another error I thought it was nicer just add `#[derive(Default)]` directly at the definition site. Technically we could remove this whole half of the macro since its only used once.

Comment (jplatte) [reply axum-core/src/macros.rs:34]: Okay, sure. 👍🏼 for getting rid of macro code that's only used once.

Comment (davidpdrsn) [reply axum-core/src/macros.rs:34]: Done!

**Review by jplatte — APPROVED** [review]
